app/controllers/user/listings_controller.rb

```diff
@@ -23,14 +25,16 @@ def create
   end
 
   def edit
-    @listing = Listing.find[:id]
+    @listing = Listing.find(params[:id])
   end
 
   def update
     @listing = Listing.find(params[:id])
-    params["listing_images"]["images"].each do |i|
-      @listing.listing_images.build(:image => i)
-    end
+     if params["listing_images"]
+       params["listing_images"]["images"].each do |i|
+       @listing.listing_images.build(image: i)
+     end
```

Comment:
`end` at 36, 5 is not aligned with `params["listing_images"]["images"].each do |i|` at 34, 7